docs/source/user_guide/memory.rst

```diff
@@ -198,3 +198,51 @@ Be sure to do this before any other CuPy operations.
 
    # Disable memory pool for pinned memory (CPU).
    cupy.cuda.set_pinned_memory_allocator(None)
+
+
+Unified memory programming (UMP) support (**experimental!**)
+............................................................
+
+It is possible to make both NumPy and CuPy use/share system allocated memory on Heterogeneous Memory Management
+(HMM) or Address Translation Services (ATS) enabled systems, such as the NVIDIA Grace Hopper Superchip.
+To activate this capability, currently you need to:
+
+1. Install `numpy_allocator <https://github.com/inaccel/numpy-allocator/>`_
+2. Set the environment variable ``CUPY_ENABLE_UMP=1``
+3. Make a memory pool for CuPy to draw system memory (``malloc_system``), for example:
+
+.. code-block:: py
+
+    import cupy as cp
+    cp.cuda.set_allocator(cp.cuda.MemoryPool(cp.cuda.memory.malloc_system).malloc)
+
+4. Switch to the aligned allocator for NumPy to draw system memory
+
+.. code-block:: py
+   
+    import cupy._core.numpy_allocator as ac
+    import numpy_allocator
```

Comment:
This doesn't work with numpy 2.0 :( Seems like they compiled it using numpy v1.x. Also when trying to build the package from scratch it resorts to numpy v1.22 for the build

Reply:
xref: https://github.com/inaccel/numpy-allocator/issues/4

Reply:
To close the loop the upstream has fixed the issue and released new wheels and conda packages! https://pypi.org/project/numpy-allocator/1.2.1/